lib/msgServer/mmrp/index.js

```diff
@@ -130,28 +147,64 @@ exports.MmrpNode = MmrpNode;
  * @param {Function} [cb]
  */
 
-MmrpNode.prototype.connect = function (uri, clusterId, cb) {
+MmrpNode.prototype.connect = function (uri, data, cb) {
+	const clusterId = data.clusterId;
+	const timestamp = data.timestamp;
+
 	// we cannot connect through routers (long routes), we are establishing a direct connection
 
 	assert.strictEqual(typeof uri, 'string');
 	assert.strictEqual(typeof clusterId, 'string');
 
-	var route = [clusterId];
-	var relay = this.relays[clusterId];
+	// check if other relay(s) with the same URI exist, and disconnect them;
```

Comment:
Except.. you don't really disconnect them. I understand you don't really want to explain that weird disconnect situation twice, but .. yeah :-/

Reply:
Suggestions?